src/main/java/org/threadly/litesockets/TCPClient.java

```diff
@@ -158,7 +159,14 @@ public void run() {
             writeFutures.clear();
             writeBuffers.discard(writeBuffers.remaining());
           }
+          if(!connectionFuture.isDone()) {
```

Comment:
Since constructed with ignoring results once complete, this check is unnecessary (and confusing IMO).